src/extractor/guidance/coordinate_extractor.cpp

```diff
@@ -337,6 +337,24 @@ util::Coordinate CoordinateExtractor::ExtractRepresentativeCoordinate(
         return result;
     }
 
+    if (IsDirectOffset(coordinates,
```

Comment:
Would you be able to add a comment that describes why `IsDirectOffset` needs to happen before `IsCurve`?  That's not obvious to me, and it seems like an important part of this change.